training/deepspeech_training/util/signal_augmentations.py

```diff
@@ -0,0 +1,200 @@
+                    self.current_sample = next_overlay_sample.audio
+                n_required = len(audio) - overlay_offset
+                n_current = len(self.current_sample)
+                if n_required >= n_current:  # take it completely
+                    overlay_data[overlay_offset:overlay_offset + n_current] += self.current_sample
+                    overlay_offset += n_current
+                    self.current_sample = None
+                else:  # take required slice from head and keep tail for next layer or sample
+                    overlay_data[overlay_offset:overlay_offset + n_required] += self.current_sample[0:n_required]
+                    overlay_offset += n_required
+                    self.current_sample = self.current_sample[n_required:]
+        snr_db = pick_value_from_range(self.snr, clock=clock)
+        orig_dbfs = max_dbfs(audio)
+        overlay_gain = orig_dbfs - max_dbfs(overlay_data) - snr_db
+        audio += overlay_data * gain_db_to_ratio(overlay_gain)
+        sample.audio = normalize_audio(audio, dbfs=orig_dbfs)
+
+    def stop(self):
+        if self.enqueue_process is not None:
+            self.enqueue_process.terminate()
+
+
+class Reverb:
+    """See "Reverb augmentation" in TRAINING.rst"""
+    def __init__(self, p=1.0, delay=20.0, decay=10.0):
+        self.probability = float(p)
+        self.delay = float_range(delay)
+        self.decay = float_range(decay)
+
+    def apply(self, sample, clock):
+        sample.change_audio_type(new_audio_type=AUDIO_TYPE_NP)
+        audio = np.array(sample.audio, dtype=np.float64)
+        orig_dbfs = max_dbfs(audio)
+        delay = pick_value_from_range(self.delay, clock=clock)
+        decay = pick_value_from_range(self.decay, clock=clock)
+        decay = gain_db_to_ratio(-decay)
+        result = np.copy(audio)
+        primes = [17, 19, 23, 29, 31]
+        for delay_prime in primes:  # primes to minimize comb filter interference
+            layer = np.copy(audio)
+            n_delay = math.floor(delay * (delay_prime / primes[0]) * sample.audio_format.rate / 1000.0)
+            n_delay = max(16, n_delay)  # 16 samples minimum to avoid performance trap and risk of division by zero
+            for w_index in range(0, math.floor(len(audio) / n_delay)):
+                w1 = w_index * n_delay
+                w2 = (w_index + 1) * n_delay
+                width = min(len(audio) - w2, n_delay)  # last window could be smaller
+                layer[w2:w2 + width] += decay * layer[w1:w1 + width]
+            result += layer
+        audio = normalize_audio(result, dbfs=orig_dbfs)
+        sample.audio = np.array(audio, dtype=np.float32)
+
+
+class Resample:
+    """See "Resample augmentation" in TRAINING.rst"""
+    def __init__(self, p=1.0, rate=8000):
+        self.probability = float(p)
+        self.rate = int_range(rate)
+
+    def apply(self, sample, clock):
+        # late binding librosa and its dependencies
```

Comment:
Why?

Reply:
It has `libsndfile` as dependency and I think it's better not to make it a requirement for everybody.